### PR TITLE
Fix broken reexport

### LINKED ISSUE #30: Both Bijectors and Distributions export "Distribution"
I'll try to send in a PR to fix this.

```
WARNING: both Bijectors and Distributions export "Distribution"; uses of it in module AdvancedVI must be qualified
ERROR: LoadError: LoadError: UndefVarError: Distribution not defined
Stacktrace:
  [1] top-level scope
    @ ~/.julia/packages/AdvancedVI/yCVq7/src/AdvancedVI.jl:96
  [2] include
    @ ./Base.jl:386 [inlined]
  [3] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1050
  [4] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:914
  [5] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:901
  [6] include
    @ ./Base.jl:386 [inlined]
  [7] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1213
  [8] top-level scope
    @ none:1
  [9] eval
    @ ./boot.jl:360 [inlined]
 [10] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [11] top-level scope
    @ none:1
in expression starting at /home/rik/.julia/packages/AdvancedVI/yCVq7/src/AdvancedVI.jl:1
in expression starting at /home/rik/.julia/packages/Turing/uMQmD/src/Turing.jl:1
```

### PR DESCRIPTION
**Pull request by rikhuijzer**
Fixes #30. **Note that the tests for AdvancedVI.jl#master are currently broken on, at least, Julia 1.6.**

This is a temporary fix for the linked issue. A full fix should either:

1. Remove `@reexport using Distributions` from Bijectors.jl (https://github.com/TuringLang/Bijectors.jl/blob/master/src/Bijectors.jl#L32)
2. Fix the underlying issue in Reexport 1.

cc @devmotion

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**/Manifest.toml
```
Comment (devmotion) [inline .gitignore:1]: Isn't
```suggestion
Manifest.toml
```
sufficient here?

**Review by devmotion — COMMENTED** [review]
I don't think this PR is a good solution. If I understand correctly it just adds a dependency that is never used explicitly? And the idea is that it avoids that users end up with Reexport 1? Unfortunately, this won't work in general, i.e., it is not guaranteed that it helps users. They might not be able to update to the new AdvancedVI version if something else is not compatible with Reexport 0.2 and/or the package resolver might not install the update since it would require to downgrade other packages.

**Comment by devmotion** [comment]
> 1. Remove `@reexport using Distributions` from Bijectors.jl (https://github.com/TuringLang/Bijectors.jl/blob/master/src/Bijectors.jl#L32)
> 2. Fix the underlying issue in Reexport 1.

I also don't think one should remove the reexports from Bijectors (it would also be a very breaking change). IMO the correct and best solution here is to fix the bugs in Reexport 1.

**Comment by rikhuijzer** [comment]
> I don't think this PR is a good solution. If I understand correctly it just adds a dependency that is never used explicitly? 

Like I've put in the comment. It's a quick fix to fix the currently broken master.

> IMO the correct and best solution here is to fix the bugs in Reexport 1.

Okay, I'll open an issue there.

**Comment by devmotion** [comment]
> Like I've put in the comment. It's a quick fix to fix the currently broken master.

I understand but as I said I don't think this PR is the correct fix here. Even if one thinks that Reexport should be bounded in a bugfix release (which I don't think it should for the reasons mentioned above) Reexport is not used by AdvancedVI so IMO it would make more sense to bound it e.g. in Bijectors where it is used.

**Comment by rikhuijzer** [comment]
> bound it e.g. in Bijectors where it is used.

Aha. Yes. You're completely right :+1:

**Comment by devmotion** [comment]
But even there I would not approve a PR :smile:

**Comment by devmotion** [comment]
If your main concern are the broken tests, an alternative might be: you could add a test/Project.toml file (instead of the extras section in Project.toml) and add Reexport there as a temporary dependency. MCMCChains (https://github.com/TuringLang/MCMCChains.jl/blob/6fc4eae9988ede0506f26bd2229c5bd99bb54518/test/runtests.jl#L1-L8) shows how it can be used also with Julia < 1.3.

**Comment by rikhuijzer** [comment]
> If your main concern are the broken tests

I mentioned the tests to allow the reader to reproduce the issue. However, the issue arises just when loading the package. That's also how I ran into it.

**Pull request closed without merging**